tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -287,21 +287,20 @@
   // Helper function to allocate a Tensor for the create() methods that create a Tensor from
   // a java.nio.Buffer.
   // Requires: dataType matches T
-  private static <T> Tensor<T> allocateForBuffer(DataType dataType, long[] shape, int nBuffered) {
-    final int nflattened = numElements(shape);
+  private static <T extends TType> Tensor<T> allocateForBuffer(DataType dataType, long[] dimSizes, int nBuffered) {
```

Comment:
Can we type `DataType` with `<T>` here? And in a bunch of other places. Or alternatively `<? extends TType>`.

Reply:
It should definitely extend `<T>`, those are left overs from previous implementation, I'm surprised my IDE is not giving me any warning about it... I'll fix it, thanks!